src/opera_disp_tms/generate_opera_disp_tile.py

```diff
@@ -0,0 +1,77 @@
+from typing import Iterable
+
+import numpy as np
+
+from opera_disp_tms.create_tile_map import create_tile_map
+from opera_disp_tms.generate_metadata_tile import create_tile_for_bbox
+from opera_disp_tms.generate_sw_disp_tile import create_sw_disp_tile
+from opera_disp_tms.generate_sw_vel_tile import create_sw_vel_tile
+
+
+def generate_opera_disp_tile(
+    tile_type: str, corner: Iterable[int], direction: str, begin_date: datetime, end_date: datetime
+):
+    bbox = [corner[0], corner[1] - 1, corner[0] + corner[1]]
+    metadata_path = create_tile_for_bbox(bbox, direction=direction)
+    if not metadata_path:
+        return
+
+    if tile_type == 'displacement':
+        out_path = create_sw_disp_tile(metadata_path, begin_date, end_date)
+    elif tile_type == 'secant_velocity':
+        out_path = create_sw_vel_tile(metadata_path, begin_date, end_date, minmax=True)
+    elif tile_type == 'velocity':
+        out_path = create_sw_vel_tile(metadata_path, begin_date, end_date, minmax=False)
+    else:
+        raise ValueError(f'Unsupported tile type: {tile_type}')
+
+    return out_path
+
+
+def generate_opera_disp_tiles(tile_type: str, bbox: Iterable[int], direction: str, begin_date: datetime, end_date: datetime):
+    lon_range = list(np.arange(bbox[0], bbox[2]))
+    lat_range = list(np.arange(bbox[1], bbox[3]))
+    tiles = []
+    for corner in combinations(lon_range, lat_range):
+        tiles.append(generate_opera_disp_tile(tile_type, corner, direction, begin_date, end_date))
+
+    scale = {
+        'displacement': None,
+        'seacant_velocity': [-0.05, 0.05],
+        'velocity': [-0.05, 0.05],
+    }
+    create_tile_map(tile_type, [str(x) for x in tiles], scale[tile_type])
+    # TOOD: upload map directory
+    # TODO: upload openlayers.html
+    # TODO: upload bounds.json
+    return Path(tile_type)
+
+
+def main():
+    """CLI entrypoint"""
+    parser = argparse.ArgumentParser(description='Create a short wavelength cumulative displacement tile')
+    parser.add_argument(
+        'tile_type', type=str, choices=['displacement', 'secant_velocity', 'velocity'], help='Type of tile to produce'
+    )
+    parser.add_argument('bbox', type=int, nargs=4, help='Upper left corner of tile in form: min_lon max_lat')
+    parser.add_argument('direction', type=str, choices=['ascending', 'descending'], help='Direction of the orbit pass')
+    parser.add_argument(
+        'begin_date', type=str, help='Start of secondary date search range to generate tile for (e.g., 20211231)'
+    )
+    parser.add_argument(
+        'end_date', type=str, help='End of secondary date search range to generate tile for (e.g., 20211231)'
+    )
+
+    args = parser.parse_args()
+    args.begin_date = datetime.strptime(args.begin_date, '%Y%m%d')
+    args.end_date = datetime.strptime(args.end_date, '%Y%m%d')
+
+    generate_opera_disp_tiles(args.tile_type, args.bbox, args.direction, args.begin_date, args.end_date)
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
```suggestion
```
Don't need these here if you're registering `opera_disp_tms.generate_opera_disp_tile.main` as an entry point because main will get called directly instead of this python file getting invoked by an interpreter; generally, for packages, only `__main__.py` would have the `if __name__...` guard.